test/bigtable/index.js

```diff
@@ -280,18 +278,28 @@ describe('Bigtable', function() {
         callback(null, response);
       };
 
+      bigtable.table = function(name) {
+        assert.strictEqual(name, fakeFormattedName);
+        return new FakeTable(bigtable, name);
```

Comment:
With this implementation, it looks like we're reliant on the functionality of `FakeTable`, which is another step in our debugging process to figure out what `FakeTable` does. I would simplify this and just return a plain `{}` and check that it's the `strictEqual` same object and had metadata set on it.